framework/src/st_things/st_things.c

```diff
@@ -403,7 +403,7 @@ int st_things_push_notification_to_cloud(const char *target_uri, st_things_repre
 	} else {
 		OCRepPayloadSetUri(resp_rep->payload, target_uri);
 		OCRepPayloadSetPropInt(resp_rep->payload, SEC_ATRRIBUTE_PUSH_MESG_MSGID, 0);
-		OCRepPayloadSetPropString(resp_rep->payload, SEC_ATRRIBUTE_PUSH_MESG_PROVIDERID, OCGetServerInstanceIDString());
+		OCRepPayloadSetPropString(resp_rep->payload, SEC_ATRRIBUTE_PUSH_MESG_PROVIDERID, (char *)OCGetServerInstanceIDString());
```

Comment:
OCGetServerInstanceIDString() returns 'const char *' and OCRepPayloadSetPropString() expects 'const char *'.
So typecast doesn't seem to be required.
Could you please re-check?